src/main/java/com/hida/MinterController.java

```diff
@@ -0,0 +1,253 @@
+    private boolean isAuto;
+    private String idType;
+    private int length;
+    private String charMap;
+
+    /**
+     * Creates a path to mint ids. If parameters aren't given then printPids
+     * will resort to using the default values found in minter_config.properties
+     *
+     * @param input - requested number of ids to mint
+     * @param model - serves as a holder for the model so that attributes can be
+     * added.
+     * @param parameters - parameters given by user to instill variety in ids
+     * @return - paths user to mint.jsp
+     */
+    @RequestMapping(value = {"/mint/{input}"},
+            method = {org.springframework.web.bind.annotation.RequestMethod.GET})
+    public String printPids(@PathVariable String input, ModelMap model,
+            @RequestParam Map<String, String> parameters) {
+
+        // message variable to be sent to mint.jsp
+        String message;
+        try {
+            int amount = Integer.parseInt(input);
+
+            System.out.print("retrieving data...");
+            // retrieve default setting
+            this.retrieveDefaultSetting();
+
+            // choose between default and given client parameters
+            this.setDefaultSetting(parameters);
+
+            // make minter object with loaded results            
+            Minter minter = new Minter(this.getCharMap(),
+                    this.getPrependedString(),
+                    amount,
+                    this.getLength(),
+                    this.getPrefix());
+
+            if (minter.getDatabaseManager().createConnection()) {
+                if (this.isAuto) {
+                    message = minter.genIdAuto(idType);
+                } else {
+                    message = minter.genIdCustom(this.isSequential);
+                }
+                minter.getDatabaseManager().closeConnection();
+                model.addAttribute("message", message);
+            } else {
+
+            }
+
+        } // detects number fomatting errors in input
+        catch (NumberFormatException exception) {
+            message = String.format(
+                    "input error %s", exception.getMessage());
+            model.addAttribute("message", message);
+        } // detects any unimplemented methods
+        catch (UnsupportedOperationException exception) {
+            message = String.format(
+                    "%s", exception.getMessage());
+            model.addAttribute("message", message);
+        } // used to see if values were properly retrieved from property file
+        catch (NullPointerException exception) {
+            System.out.println(String.format("prepend=%s\nprefix=%s\n"
+                    + "length=%d\ncharMap=%s\nisAuto=%b\n"
+                    + "isSequential=%b\nidType=%s", prependedString, prefix,
+                    length, charMap, isAuto, isSequential, idType));
+            System.out.println(Arrays.toString(exception.getStackTrace()));
+        }
+
+        return "mint";

```

Comment:
Let's return the raw `JSON` list of IDs here instead of redirecting to a JSP page.